packages/react-devtools-extensions/src/astUtils.js

```diff
@@ -29,9 +29,25 @@ const AST_NODE_TYPES = Object.freeze({
 });
 
 // Check if line number obtained from source map and the line number in hook node match
-function checkNodeLocation(path: NodePath, line: number): boolean {
+function checkNodeLocation(
+  path: NodePath,
+  line: number,
+  column: number,
+): boolean {
   const {start, end} = path.node.loc;
-  return line >= start.line && line <= end.line;
+
+  if (line < start.line || line > end.line) {
+    return false;
+  }
+
+  if (
+    (line === start.line && column < start.column) ||
+    (line === end.line && column > end.column)
```

Comment:
If I'm not mistaken, `column` is 1-based here (parsed from `Error.prototype.stack`) while `start.column` and `end.column` are 0-based (coming from the AST).

Reply:
Hey @motiz88 Thanks for checking this PR out.

Line number and column number typically come from the 'source-map' library, but they sometimes come from the 'error-stack-parser' library (if there's no source map):
https://github.com/facebook/react/blob/cb8afda183e9c931978279d3a1706d1d9c905484/packages/react-devtools-extensions/src/parseHookNames.js#L370-L387

Would we need to add 1 when the values come from the 'source-map' library (so that we could later subtract one here)?

Ideally we'd write a unit test that caught these off-by-one errors to prevent from regressing.

Reply:
> If I'm not mistaken, `column` is 1-based here (parsed from `Error.prototype.stack`) while `start.column` and `end.column` are 0-based (coming from the AST).

Just confirmed this using AST Explorer.

Reply:
I think...in practice...this will never cause a problem, because we are matching the 1-based Error stack location for the hook `Identifier` (e.g. `useState`) against the 0-based larger `VariableDeclarator` (e.g. `[foo, setFoo] = useState()`) so the ranges will always overlap.

I think we should still fix the code, but I'm not sure if it's possible to write an integration test that would fail without this change.

Reply:
#21865